packaging/package-indexer/indexer/rpm_indexer.py

```diff
@@ -0,0 +1,304 @@
+#############################################################################
+# Copyright (c) 2024 Axoflow
+# Copyright (c) 2024 Attila Szakacs <attila.szakacs@axoflow.com>
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of the GNU General Public License version 2 as published
+# by the Free Software Foundation, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
+#
+# As an additional exemption you are allowed to compile & link against the
+# OpenSSL libraries as published by the OpenSSL project. See the file
+# COPYING for details.
+#
+#############################################################################
+
+import shutil
+import re
+from pathlib import Path
+from tempfile import TemporaryDirectory, TemporaryFile, NamedTemporaryFile
+from typing import List, Optional
+
+from cdn import CDN
+from remote_storage_synchronizer import RemoteStorageSynchronizer
+
+from indexer import Indexer
+
+from . import utils
+
+CURRENT_DIR = Path(__file__).parent.resolve()
+
+
+class RpmIndexer(Indexer):
+    def __init__(
+        self,
+        incoming_remote_storage_synchronizer: RemoteStorageSynchronizer,
+        indexed_remote_storage_synchronizer: RemoteStorageSynchronizer,
+        incoming_sub_dir: Path,
+        dist_dir: Path,
+        cdn: CDN,
+        gpg_key_path: Path,
+        gpg_key_passphrase: Optional[str],
+    ) -> None:
+        self.__gpg_key_path = gpg_key_path
+        self.__gpg_key_passphrase = gpg_key_passphrase
+        super().__init__(
+            incoming_remote_storage_synchronizer=incoming_remote_storage_synchronizer,
+            indexed_remote_storage_synchronizer=indexed_remote_storage_synchronizer,
+            incoming_sub_dir=incoming_sub_dir,
+            indexed_sub_dir=Path("rpm", dist_dir),
+            cdn=cdn,
+        )
+
+    def __move_files_from_incoming_to_indexed(self, incoming_dir: Path, indexed_dir: Path) -> None:
+        for file in filter(lambda path: path.is_file(), incoming_dir.rglob("*")):
+            relative_path = file.relative_to(incoming_dir)
+            platform = relative_path.parent
+            file_name = relative_path.name
+
+            new_path = Path(indexed_dir, platform, "x86_64", file_name)
```

Comment:
Can we extract the arch string to a global constant? (We are not far away from supporting aarch64)

Reply:
Yes, done.